beacon-chain/blockchain/execution_engine.go

```diff
@@ -265,7 +265,8 @@ func (s *Service) getPayloadAttribute(ctx context.Context, st state.BeaconState,
 
 	// Get previous randao.
 	st = st.Copy()
-	st, err := transition.ProcessSlotsIfPossible(ctx, st, slot)
+	headRoot := s.headRoot()
+	st, err := transition.ProcessSlotsUsingNextSlotCache(ctx, st, headRoot[:], slot)
```

Comment:
There is a functional difference between `ProcessSlotsIfPossible` and using NSC: the former does not error out if the requested slot is higher than the current slot. I've seen this fail on tests and I wonder if this won't fail on runtime.

Reply:
Right. The easiest way to solve this is to add process slots only if it's higher within `ProcessSlotsUsingNextSlotCache`. It seems no harm to do that but curious what you think